src/ImageSharp/Processing/Processors/AdaptiveThresholdProcessor.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing.Processors;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    /// <summary>
+    /// Performs Bradley Adaptive Threshold filter against an image
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format</typeparam>
+    internal class AdaptiveThresholdProcessor<TPixel> : ImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        public AdaptiveThresholdProcessor() { }
+
+        protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle sourceRectangle, Configuration configuration)
+        {
+            // What does this do?
+            // int startY = sourceRectangle.Y;
+            // int endY = sourceRectangle.Bottom;
+            // int startX = sourceRectangle.X;
+            // int endX = sourceRectangle.Right;
+            // int minX = Math.Max(0, startX);
+            // int maxX = Math.Min(source.Width, endX);
+            // int minY = Math.Max(0, startY);
+            // int maxY = Math.Min(source.Height, endY);
+
+            /*
+                The implementation here does not account for Rectangles
+                or regions of the image. Needs to be implemented
```

Comment:
Use `Rectangle.Intersect` and then use the bounds in place of height/width below.